tensorflow/contrib/avro/python/tests/parse_avro_record_test.py

```diff
@@ -0,0 +1,1143 @@
+                      'list_of_records/[0]/first_name': parsing_ops.FixedLenFeature([], tf_types.string),
+                      "map_of_records/['second']/age": parsing_ops.FixedLenFeature([], tf_types.int32)},
+            should_pass=True
+        ))
+
+    def test_nested_with_asterisk(self):
+        """
+        Test nested records with the asterisk notation. In this case we want all items from an array/map.
+        """
+        self.run_test(DataForTest(
+            schema='''{"doc": "Nested records in array to use asterisk.",
+                      "namespace": "com.test.nested.records",
+                      "type": "record",
+                      "name": "test_nested_record",
+                      "fields": [
+                         {"name": "index", "type": "int"},
+                         {"name": "sparse_type",
+                            "type":
+                            {
+                                "type": "array",
+                                "items": {
+                                    "type": "record",
+                                    "name": "sparse_triplet",
+                                    "fields": [
+                                        {"name": "index", "type": "long"},
+                                        {"name": "max_index", "type": "int"},
+                                        {"name": "value", "type": "float"}
+                                    ]
+                                }
+                            }
+                        }]}''',
+            data=[{'index': 0, 'sparse_type': [{'index': 0, 'max_index': 10, 'value': 5.0},
+                                               {'index': 5, 'max_index': 10, 'value': 7.0},
+                                               {'index': 3, 'max_index': 10, 'value': 1.0}]},
+                  {'index': 1, 'sparse_type': [{'index': 0, 'max_index': 10, 'value': 2.0},
+                                               {'index': 9, 'max_index': 10, 'value': 1.0}]}],
+            features={'index': parsing_ops.FixedLenFeature([], tf_types.int32),
+                      'sparse_type/[*]/index': parsing_ops.VarLenFeature(tf_types.int64),
+                      'sparse_type/[*]/value': parsing_ops.VarLenFeature(tf_types.float32)},
+            should_pass=True))
+
+    def test_parse_int_as_long_fail(self):
+        """
+        Test a failure case where we want to parse an int as a long value.
+        """
+        self.run_test(DataForTest(
+            schema='''{"doc": "Parse int as long (int64) fails.",
+                       "namespace": "com.test.int.type.failure",
+                       "type": "record",
+                       "name": "data_row",
+                       "fields": [
+                         {"name": "index", "type": "int"}
+                       ]}''',
+            data=[{'index': 0}],
+            features={'index': parsing_ops.FixedLenFeature([], tf_types.int64)},
+            should_pass=False
```

Comment:
Consider adding an exception type or error message to expect.

Reply:
Design changed. Please let me know if it would still be beneficial having a separate exception type?